frontend/app_flowy/lib/common/theme/theme.dart

```diff
@@ -0,0 +1,151 @@
+import 'package:flutter/material.dart';
+
+// FIXME: Different colors for different themes (light/dark).
+extension ColorSchemeExtension on ColorScheme {
+  Color get red => const Color(0xFFffadad);
+  Color get orange => const Color(0xFFffcfad);
+  Color get yellow => const Color(0xFFfffead);
+  Color get lime => const Color(0xFFe6ffa3);
+  Color get green => const Color(0xFFbcffad);
+  Color get aqua => const Color(0xFFadffe2);
+  Color get blue => const Color(0xFFade4ff);
+  Color get purple => const Color(0xFFc3adff);
+  Color get pink => const Color(0xFFffadf9);
+
+  Color get shader1 => _isDark ? const Color(0xFFffffff) : const Color(0xFF333333);
+  Color get shader2 => _isDark ? const Color(0xFFffffff) : const Color(0xFF4f4f4f);
+  Color get shader3 => const Color(0xFF828282);
+  Color get shader4 => const Color(0xFFbdbdbd);
+
+  bool get _isDark => brightness == Brightness.dark;
+}
+
+class FlowyTheme {
+  FlowyTheme(this._theme);
+
+  FlowyTheme.fromName(String name) {
+    _theme = DefaultThemesExtension.fromName(name);
+  }
+
+  late final DefaultThemes _theme;
+
+  DefaultThemes get theme => _theme;
+
+  bool get isDark => _theme == DefaultThemes.dark ? true : false;
+
+  ThemeData get themeData => ThemeData(
+        brightness: isDark ? Brightness.dark : Brightness.light,
+
+        primaryColor: const Color(0xFF00bcf0),
+
+        // TODO: Custom color with tins and shades
+        primarySwatch: Colors.blue,
+
+        scaffoldBackgroundColor: isDark ? const Color(0xFF292929) : const Color(0xFFf7f8fc),
```

Comment:
How can I extend the others theme type? For example, we have an enum for ThemeType.
enum ThemeType {
  light,
  dark,
}

Reply:
Yes we could get rid of the `isDark` (or `brightness`) since there will be more themes than just `light` and `dark`, so I think this `enum ThemeType` should becomes `class ThemeType extends ThemeData`, so we could reload `ThemeData` programmatically when user change the theme and reusing the `properties`.

Hmmm, so because of that I don't think what you want to do have any problem with this PR so far. This PR is solely for refactoring `AppTheme` to `Theme.of(context)`.

But I understand what you wanted and I don't have enough knowledge of how Flutter could load JSON (https://github.com/peiffer-innovations/json_theme) at runtime (using Rust instead?). But if we could found a way to load JSON then reloading `ThemeData` is not really a big deal with `ThemeCubit`.